### PR TITLE
Feature/mcuboot single image dfu

### PR DESCRIPTION
**Pull request by de-nordic**
Commit `[nrf fromlist] zephyr: Allow single image applications` is also reviewed here:
~~https://github.com/zephyrproject-rtos/mcuboot/pull/18~~
https://github.com/JuulLabs-OSS/mcuboot/pull/741
https://github.com/JuulLabs-OSS/mcuboot/pull/752

West update here:
https://github.com/nrfconnect/sdk-nrf/pull/2291

### CODE REVIEW

**Comment by de-nordic** [comment]
The DT macros that pick flash image differ between this and upstream pull request as the upstream already uses new DT API.

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -393,6 +405,13 @@ config RECOVERY_UART_DEV_NAME
 
 endif # MCUBOOT_SERIAL
 
+config UPDATEABLE_IMAGE_NUMBER
```
Comment (nvlsianpu) [inline boot/zephyr/Kconfig:408]: Please include copy of https://github.com/JuulLabs-OSS/mcuboot/commit/16b6d15a6b0dc8fe723b47e8fcf5f742f4f002e6
This option is duplicated in current patch version.

**Comment by nvlsianpu** [comment]
Remove [DNM] as patch is already accepted upstream.

**Comment by de-nordic** [comment]
No longer needed.

**Pull request closed without merging**